hack/prepare-release.sh

```diff
@@ -0,0 +1,161 @@
+        if [[ ! -z "${BASH_REMATCH[5]}" ]]; then
+            prerelease=${BASH_REMATCH[5]}
+            prerelease_regex="^([0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)$"
+            if ! [[ $prerelease =~ $prerelease_regex ]]; then
+                echo "Invalid SemVer format: $version"
+                return 1
+            fi
+        fi
+
+        echo "Valid SemVer format: $version"
+        return 0
+    else
+        echo "Invalid SemVer format: $version"
+        return 1
+    fi
+}
+
+#
+# Parse command line
+#
+no_patching=
+previous_version=$(git describe --tags $(git rev-list --tags --max-count=1))
+# Parse command line options
+while [[ $# -gt 0 ]]; do
+    key="$1"
+    case $key in
+        --no-patch)
+            no_patching=1
+            shift
+            ;;
+        --previous-version)
+            previous_version="$2"
+            shift 2
+            ;;
+        --help/-h)  usage
+            exit 0
+            ;;
+        *) break
+            ;;
+    esac
+done
+shift "$((OPTIND - 1))"
+
+# Check that no extra args were provided
+if [ $# -ne 1 ]; then
+    if [ $# -lt 1 ]; then
+        echo -e "ERROR: too few arguments\n"
+    else
+        echo -e "ERROR: unknown arguments: ${@:3}\n"
+    fi
+    usage
+    exit 1
+fi
+
+release=$1
+shift 1
+
+container_image=nvcr.io/nvidia/k8s-device-plugin:$release
+
+#
+# Check/parse release number
+#
+if [ -z "$release" ]; then
+    echo -e "ERROR: missing RELEASE_VERSION\n"
+    usage
+    exit 1
+fi
+
+# validate the release version
+if ! validate_semver $release; then
+    echo -e "ERROR: invalid RELEASE_VERSION\n"
+    exit 1
+fi
+semver=${release:1}
+
+# validate the previous version
+if ! validate_semver $previous_version; then
+    echo -e "ERROR: invalid PREVIOUS_VERSION\n"
+    exit 1
+fi
+pre_semver=${previous_version:1}
+
+#
+# Modify files in the repo to point to new release
+#
+if [ -z "$no_patching" ]; then
+    # Patch versions.mk
+    echo Patching versions.mk to refer to $release
+    sed -i '' "s/^VERSION.*$/VERSION ?= $release/" versions.mk
+
+    # Patch deployments/static
+    echo Patching deployments to refer to $semver
+    find deployments/static -type f \( -name "*.yaml" -o -name "*.yml" \) -exec sed -E -i '' \
+        -e s",^([[:space:]]+)app.kubernetes.io\/version:.+$,\1app.kubernetes.io\/version: $semver," \
+        -e s",^([[:space:]]+)- image:.+$,\1- image: $container_image," \
+        {} +
+    
+    # Patch deployments/helm/Chart.yaml
+    echo Patching deployments/helm/Chart.yaml to refer to $semver
+    sed -i '' "s/^version: .*/version: \"$semver\"/" deployments/helm/nvidia-device-plugin/Chart.yaml
+    sed -i '' "s/^appVersion: .*/appVersion: \"$semver\"/" deployments/helm/nvidia-device-plugin/Chart.yaml
+
+    # Patch README.md
+    # NOTE: this needs to be updated once we start releasing x.2.x versions
+    echo Patching README.md to refer to $release
+    sed -E -e "s/([^[:space:]])$previous_version([^[:alnum:]]|$)/\1$release\2/g" -i '' README.md
```

Comment:
Nit. Can we be consistent with the other commands in terms of where the `-i` flag is.